### PR TITLE
[CARBONDATA-3158] support presto-carbon to read sdk cabron files

### PR DESCRIPTION
**Pull request by ajantha-bhat**
**problem:**
Currently, carbon SDK files output (files without metadata folder and its
contents) are read by spark using an external table with carbon session.
But presto carbon integration doesn't support that. It can currently read
only the transactional table output files.

**solution:** 
Hence we can enhance presto to read SDK output files. This will increase
the use cases for presto-carbon integration.

The above scenario can be achieved by inferring schema if metadata folder
not exists and
setting read committed scope to LatestFilesReadCommittedScope, if
non-transctional table output files are present.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? NA
 
 - [ ] Any backward compatibility impacted? NA
 
 - [ ] Document update required? NA

 - [ ] Testing done.  
        yes, added UT       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1681/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9941/

File: integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java [hunk 1]
```diff
@@ -364,23 +355,38 @@ private CarbonTable parseCarbonMetadata(SchemaTableName table) {
...
-        // TBase is used to read and write thrift objects.
-        // TableInfo is a kind of TBase used to read and write table information.
-        // TableInfo is generated by thrift,
-        // see schema.thrift under format/src/main/thrift for details.
-        public TBase create() {
-          return new org.apache.carbondata.format.TableInfo();
+      String metadataPath = CarbonTablePath.getSchemaFilePath(tablePath);
+      boolean isTransactionalTable = false;
+      try {
+        if (FileFactory.getCarbonFile(metadataPath)
```
Comment (qiuchenjian) [inline integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java]: The operator of getFileType is called twice in getCarbonFile and if(...), it's better to getting it ahead 
FileType fileType = FileFactory.getFileType(metadataPath);
if (FileFactory.getCarbonFile(metadataPath, fileType).isFileExist(metadataPath,fileType)) {...}

File: integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java [hunk 2]
```diff
@@ -364,23 +355,38 @@ private CarbonTable parseCarbonMetadata(SchemaTableName table) {
...
+          }
+        };
+        ThriftReader thriftReader =
+            new ThriftReader(CarbonTablePath.getSchemaFilePath(tablePath), createTBase);
+        thriftReader.open();
+        tableInfo = (org.apache.carbondata.format.TableInfo) thriftReader.read();
+        thriftReader.close();
+      } else {
+        tableInfo =
+            CarbonUtil.inferSchema(tablePath, table.getTableName(), false, new Configuration());
```
Comment (qiuchenjian) [inline integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java]: Is this code (tableInfo = CarbonUtil.inferSchema(tablePath, table.getTableName(), false, new Configuration());)  tested on hdfs,  FileSystem may be not created by "new Configuration()"

Comment (ajantha-bhat) [reply integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java]: I have tested. It works.

but better to use
FileFactory.getConfiguration(). I will change to it,

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1697/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1907/

**Comment by ajantha-bhat** [comment]
@ravipesala , @jackylk : PR is ready. please review

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9957/

File: integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java [hunk 3]
```diff
@@ -364,23 +355,38 @@ private CarbonTable parseCarbonMetadata(SchemaTableName table) {
...
-        // TableInfo is a kind of TBase used to read and write table information.
-        // TableInfo is generated by thrift,
-        // see schema.thrift under format/src/main/thrift for details.
-        public TBase create() {
-          return new org.apache.carbondata.format.TableInfo();
+      String metadataPath = CarbonTablePath.getSchemaFilePath(tablePath);
+      boolean isTransactionalTable = false;
+      try {
+        FileFactory.FileType fileType = FileFactory.getFileType(metadataPath);
+        if (FileFactory.getCarbonFile(metadataPath, fileType).isFileExist(metadataPath, fileType)) {
```
Comment (ravipesala) [inline integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java]: No need to pass filetype here

**Comment by ravipesala** [comment]
LGTM , just a minor comment.

File: integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java (hunk 3 above)
Comment (ajantha-bhat) [reply integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java]: currently, this reduces one function call, If I don't pass file type implicitly again they call this method. Let it be there now. 

when we remove filetype from file factory in the future, this also will be optimized.

**Pull request closed without merging**